### PR TITLE
chore(hana-client): add warning for missing thread pool configuration

### PR DESCRIPTION
**Pull request by David-Kunz**
Also asked for review in mail thread

### CODE REVIEW

**Review by BobdenOs — APPROVED** [review]

**Review by danjoa — APPROVED** [review]
Looks good 👍 
Could we add a learn more link to some docs by hana-client?

File: hana/lib/drivers/hana-client.js [hunk 1]
```diff
@@ -17,6 +19,7 @@ class HANAClientDriver extends driver {
    * @param {import('./base').Credentials} creds The credentials for the HANAClientDriver instance
    */
   constructor(creds) {
+    if (process.env.NODE_ENV === 'production' && !process.env.HDB_NODEJS_THREADPOOL_SIZE && !process.env.UV_THREADPOOL_SIZE) LOG.warn("When using @sap/hana-client, it's strongly recommended to adjust its thread pool size with environment variable `HDB_NODEJS_THREADPOOL_SIZE`, otherwise it might lead to performance issues.")
```
Comment (BobdenOs) [inline hana/lib/drivers/hana-client.js:22]: I think we should remove the `NODE_ENV` check and move it to the root of the file. So it only is printed on the first time connecting to the database. This file is only loaded when `@sap/hana-client` is actually used. That way it is not printed once.

Comment (David-Kunz) [reply hana/lib/drivers/hana-client.js:22]: Sure, I can move everything

Comment (David-Kunz) [reply hana/lib/drivers/hana-client.js]: done

**Pull request merged**